### ISSUE TITLE
content-directory-override / shader options don't work on subfolders

### ISSUE THREAD

**Issue by SiriusVI87** [libretro/RetroArch#16993]
Example:

I have a DOS games launched with the dosbox core and also windows 3.1 games that are launched with the dosbox core. I want to use different shader presets for dos games and windows 3.1 games. Both platforms have their games stored in individual folders.

The folder structure is as follows:

MS DOS
 -- game folder 1
 -- game folder 2
 -- ...

Windows 3.1
 -- game folder 1
 -- game folder 2
 -- ...
 
So since both platforms use the same core, I cannot use the save shaders for the whole core option. But I cannot use the "content directory" option either, because that does not work for subfolders.

What I'd like to see is an option in RetroArch that allows me to apply teh "content directory" option for all the subfolders of the content directory folder as well.

Is this possible?

Thx!

**Comment by acschwartz** [comment]
Seconding this request; it makes no sense that content directory overrides are not recursive

**Comment by hizzlekizzle** [comment]
If they were recursive, we would have to implement an entire hierarchy for dealing with conflicting overrides. For example, in structure foo/bar, you could have an override for foo, which would also apply to bar, but then you could have one for bar, and then additional ones for any subdir under that.

So, any time you loaded something, it would have to identify, parse, and reconcile all of those recursive, conflicting overrides.

**Comment by acschwartz** [comment]
I don't think the conflicting overrides need to be reconciled for this feature to work in a way that makes sense.
When dealing with Game, Content Directory, and Core overrides, the 'winning' override completely overrides the lower-priority ones, does it not?

So for `/foo/bar/`, check for Content Directory override exists for `bar`. If it exists, use it without reconciling (I believe this is how overrides work now?). If it does not exist, use the content directory override for its parent directory. If parent directory does not have an override, continue traveling up the parent directory chain until it's determined there are no Content Directory overrides that would apply. Then check for other lower-priority overrides as you normally would. 

(Btw, thanks for taking the time to respond! I found this issue from a reddit discussion of this feature, and it's nice to at least get an explanation of why it hasn't been implemented)

**Comment by hizzlekizzle** [comment]
Right, only one override applies. By "reconcile," I meant: it has to figure out which one has primacy among them, comparing the path and where the match (if there is one) exists inside that path.

It can be done--it's software; pretty much anything can be done with the right motivation--but it's messy.
